src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -869,11 +869,25 @@ public String waitForCompletion(int nodeId,
 
             String description = null;
             String oldStatus = "";
+
+            // State to detect hung async jobs
+            long oldStatusTime = -1;
+            long timeSinceLastWarning = -1;
```

Comment:
The variable is updated whenever status is reported, or warning is reported. So I am not sure the warn should be in the name. This is  a very minor detail, so I would leave it to you which way you want to go with.